src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -119,6 +121,12 @@ public DatasetService(
   public String createDataset(
       DatasetRequestModel datasetRequest, AuthenticatedUserRequest userReq) {
     String description = "Create dataset " + datasetRequest.getName();
+    HashMap<String, Object> properties =
+        new HashMap<>(
+            Map.of(
+                BardEventProperties.BILLING_PROFILE_ID_FIELD_NAME,
+                datasetRequest.getDefaultProfileId()));
+    UserMetricsInterceptor.eventProperties.set(properties);
```

Comment:
Just to make sure I'm understanding --- I know this ticket is specifically requesting that we count the number of datasets created (& record the associated billing profile). So, if we want to also record the number of snapshot created, we could just copy this chunk of code over the snapshot create flight, right? We could reuse the same bard event property.

Reply:
Yeah that's right!